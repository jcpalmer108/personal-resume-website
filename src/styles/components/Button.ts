import styled from "styled-components";
import { desktopBreakpoint, redstone, tabletBreakpoint } from "../variables";

export const Link = styled.a`
  width: 186px;
  height: 55px;
  display: flex;
  flex-direction: row;
  align-items: center;
  column-gap: 10px;
  background-color: ${redstone};
  justify-content: center;
  text-decoration: none;
  color: white;
  font-family: Gilroy-Medium;
  font-size: 12px;
  transition: all 500ms ease;

  @media (min-width: ${tabletBreakpoint}px) {
    width: 214px;
    height: 60px;
    font-size: 14px;
    border: 1px solid transparent;
    transition: all 500ms ease;

    &:hover {
      color: ${redstone};
      background: white;
      transition: all 500ms ease;
      font-size: 16px;
      border: ${redstone} 1px solid;

      img {
        filter: invert(59%) sepia(50%) saturate(1158%) hue-rotate(312deg)
          brightness(92%) contrast(95%);
        transition: all 500ms ease;
      }
    }
  }

  @media (min-width: ${desktopBreakpoint}px) {
    width: 218px;
    height: 64px;
  }
`;

export const ArrowRight = styled.img`
  width: 22px;
  padding-bottom: 3px;
`;
