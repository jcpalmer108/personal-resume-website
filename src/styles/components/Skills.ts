import styled from "styled-components";
import { contentWidth, desktopBreakpoint, silver_polish, tabletBreakpoint } from "../variables";

export const Mobile = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  text-align: center;
  padding-bottom: 40px;

  @media (min-width: ${tabletBreakpoint}px) {
    display: none;
  }
`

export const Title = styled.h2`
  padding: 15px 15px 30px 15px;
  max-width: 250px;

  @media (min-width: ${tabletBreakpoint}px) {
    max-width: none;
    padding-top: calc(7% - 25px);
    padding-left: 25px;
    position: absolute;
    width: calc(60% - 50px);
    max-width: 500px;
    overflow: hidden;
  }

  @media (min-width: ${desktopBreakpoint}px) {
    padding-top: 45px;
    padding-left: 60px;
    width: 600px;
  }

`

export const Logo = styled.img`
  height: 50px;
  width: 50px;
  object-fit: cover;

  @media (min-width: ${tabletBreakpoint}px) {
    height: 70px;
    width: 70px;
  }

  @media (min-width: ${desktopBreakpoint}px) {
    height: 100px;
    width: 100px;
  }


  `

export const Link = styled.a`
  display: flex;
  width: 50px;
  height: 50px;
  border: 1px solid ${silver_polish};
  padding: 30px;
  margin-right: 13px;

  @media (min-width: ${tabletBreakpoint}px) {
    padding: 0px;
    position: absolute;
    border: none;
    margin-right: 0px;
    padding-bottom: 0px;
    
    max-width: 230px;
    align-items: center;
    justify-content: center;
    width: 20%;
    padding-top: 7%;
    max-height: 200px;
  }

  @media(min-width: ${desktopBreakpoint}px) {
    padding-top: 8%;
  }

  @media (min-width: ${contentWidth}px) {
    padding-top: 90px;
  }


`

export const Description = styled.div`
  padding: 30px 20px 50px 20px;
  font-family: Gilroy-Regular;

  @media (min-width: ${tabletBreakpoint}px) {
    padding: 20px 0px 30px 0px;
    width: 100%;
  }

  @media (min-width: ${desktopBreakpoint}px) {
    padding: 7% 0px 30px 0px;
    width: 145%;
  }
`

export const Paragraph = styled.p`
  margin-bottom: 16px;
  margin: 0px;

  &:last-child {
    margin-bottom: 0px;
  }

  @media (min-width: ${tabletBreakpoint}px) {
    position: relative;
  }

`

export const NotMobile = styled.div`
  display: none;
  
  @media (min-width: ${tabletBreakpoint}px) {
    display: flex;
    flex-direction: column;
  }

`

export const Mosaic = styled.table`

  @media(min-width: ${tabletBreakpoint}px) {
    width: 100%;
    border-collapse: collapse;
  }
`

type ContentProps = {
  desktop?: boolean | undefined
}

export const Content = styled.div<ContentProps>`
  height: 0px;
  width: 60%;
  max-width: 600px;
  display: block;

  @media (min-width: ${tabletBreakpoint}px) {
    display: ${ props => props.desktop ? "none" : "block"};
    padding: 25px;
    height: 100%;
    max-width: none;
    width: calc(100% - 50px);
  }

  @media (min-width: ${desktopBreakpoint}px) {
    display: ${ props => props.desktop ? "block" : "none"};
    position: relative;
    height: 0px;
    padding: 0px;
    width: 60%;
    margin-left: 60px;
    margin-right: 60px;
  }
  `


type CellProps = {
  first?: boolean | undefined,
  last?: boolean | undefined
}
export const Cell = styled.td<CellProps>`

  @media(min-width: ${tabletBreakpoint}px) {
    border-width: 1px;
    border-left: ${props => props.first ? 'none' : 'solid'};
    border-top: ${props => props.first ? 'none' : 'solid'};
    border-bottom: ${props => props.last ? 'none' : 'solid'};
    border-right: ${props => props.last ? 'none' : 'solid'};
    border-color: ${silver_polish};

    width: 20%;
    height: 0px;
    padding-bottom: 20%;
  }

  @media(min-width: ${contentWidth}px) {
    &:first-child {
      border-left: none;
    }

    &:last-child {
      border-right: none;
    }
  }
  `
