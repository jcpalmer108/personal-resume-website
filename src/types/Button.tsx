export type ButtonProps = {
  label: string;
  url?: string;
  onClick?: Function;
};
