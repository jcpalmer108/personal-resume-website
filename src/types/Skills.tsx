import { Sections } from "./content"

export type SkillsProps = {
  content: Sections | undefined
}