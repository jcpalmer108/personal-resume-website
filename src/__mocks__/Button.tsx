import React from "react";
import { ButtonProps } from "../types/Button";

export function Button({ label, url }: ButtonProps) {
  return (
    <div>
      {label}
      {url}
    </div>
  );
}
