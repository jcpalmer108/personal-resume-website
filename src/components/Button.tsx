import arrow from "../assets/images/icons/arrow-right.svg";
import { ArrowRight, Link } from "../styles/components/Button";
import { ButtonProps } from "../types/Button";

export default function Button({ label, url, onClick }: ButtonProps ) {
  if(!url && !onClick) {
    return null;
  }

  return (
    <div data-testid="Button" onClick={() => { if(onClick) onClick() }}>
      <Link href={url}>
        <span>{label.toUpperCase()}</span>
        <ArrowRight src={arrow} alt="arrow" />
      </Link>
    </div>
  )
}