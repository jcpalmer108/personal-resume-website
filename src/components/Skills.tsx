import { useState } from "react";
import Marquee from "react-fast-marquee";
import Section from "../components/Section";
import {
  ButtonWrapper,
  Cell,
  Content,
  Description,
  Link,
  Logo,
  Mobile,
  Mosaic,
  NotMobile,
  Title,
} from "../styles/components/Skills";
import { Skill as SkillsContent } from "../types/content";
import {
  DescriptionAndActionProps,
  SkillsProps,
  TileProps,
  TitleProps,
} from "../types/Skills";
import Button from "./Button";
import SkillsModal from "./SkillsModal";

export default function Skills({ content }: SkillsProps) {
  const skillList = content?.subSection?.skills?.data.filter(
    (item) => item.url && item.category !== "soft-skills"
  );
  const title = content?.title;
  const description = (content?.description && content?.description[0]) || "";
  const [modalIsOpen, setModalIsOpen] = useState(false);

  const generateTiles = (
    tiles: SkillsContent[] | undefined,
    table: boolean = true
  ) => {
    const content = tiles ? tiles : [];
    return content.map((item, index) => (
      <Tile key={`Tile ${index + 1}`} info={item} table={table} />
    ));
  };

  if (skillList && skillList.length < 9) {
    return null;
  }

  return (
    <div id="skills">
      {modalIsOpen && (
        <SkillsModal
          closeModal={() => setModalIsOpen(!modalIsOpen)}
          content={content?.subSection?.skills?.data}
          categories={content?.subSection?.skills?.categories}
        />
      )}
      <Section wide center label={content?.label}>
        <Mobile data-testid="Mobile">
          <TitleLine title={title || ""} />
          <Marquee loop={0}>{generateTiles(skillList, false)}</Marquee>
          <DescriptionAndAction
            description={description || ""}
            openModal={() => setModalIsOpen(true)}
          />
        </Mobile>
        <NotMobile data-testid="NotMobile">
          <Mosaic>
            <tbody>
              <tr data-testid="Row 1">
                <Cell colSpan={3} first>
                  <TitleLine title={title || ""} />
                </Cell>
                {generateTiles(skillList?.slice(0, 2))}
              </tr>
              <tr data-testid="Row 2">
                {generateTiles(skillList?.slice(2, 7))}
              </tr>
              <tr data-testid="Row 3">
                {generateTiles(skillList?.slice(7, 9))}
                <Cell colSpan={3} last>
                  <Content desktop>
                    <DescriptionAndAction
                      description={description || ""}
                      openModal={() => setModalIsOpen(true)}
                    />
                  </Content>
                </Cell>
              </tr>
            </tbody>
          </Mosaic>
          <Content>
            <DescriptionAndAction
              description={description || ""}
              openModal={() => setModalIsOpen(true)}
            />
          </Content>
        </NotMobile>
      </Section>
    </div>
  );
}

function Tile({ info, table }: TileProps) {
  const { key, url, label } = info;

  return table === true ? (
    <Cell>
      <Link id={key} href={url} data-testid="Tile">
        <Logo
          src={require("../assets/images/skills/section/" + key + ".svg")}
          alt={label}
        />
      </Link>
    </Cell>
  ) : (
    <Link id={key} href={url}>
      <Logo
        src={require("../assets/images/skills/section/" + key + ".svg")}
        alt={label}
      />
    </Link>
  );
}

function DescriptionAndAction({
  openModal,
  description,
}: DescriptionAndActionProps) {
  return (
    <>
      <Description data-testid="Description">{description}</Description>
      <ButtonWrapper>
        <Button onClick={() => openModal()} label="Let's Talk" />
      </ButtonWrapper>
    </>
  );
}

function TitleLine({ title }: TitleProps) {
  return <Title data-testid="Title">{title}</Title>;
}
