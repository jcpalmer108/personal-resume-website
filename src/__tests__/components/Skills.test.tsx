import { render, screen } from '@testing-library/react';
import 'jest-styled-components'
import Skills from '../../components/Skills';
import Button from "../../components/Button";

jest.mock('../../components/Button')

const params = {
  key: "about",
  label: "Skills",
  inMenu: true,
  title: "test title",
  description: [
    "test contents 1",
    "test contents 2"
  ],
  subSection: {
    skills: [
      {
        key: "github",
        label: "Test Skill 1",
        url: "https://test.skill.one/"
      },
      {
        key: "github",
        label: "Test Skill 2",
        url: "https://test.skill.two/"
      },
      {
        key: "github",
        label: "Test Skill 3",
        url: "https://test.skill.three/"
      },
      {
        key: "github",
        label: "Test Skill 4",
        url: "https://test.skill.four/"
      },
      {
        key: "github",
        label: "Test Skill 5",
        url: "https://test.skill.five/"
      },
      {
        key: "github",
        label: "Test Skill 6",
        url: "https://test.skill.six/"
      },
      {
        key: "github",
        label: "Test Skill 7",
        url: "https://test.skill.seven/"
      },
      {
        key: "github",
        label: "Test Skill 8",
        url: "https://test.skill.eight/"
      },
      {
        key: "github",
        label: "Test Skill 9",
        url: "https://test.skill.nine/"
      }
    ]
  }
}

describe('Skills', () => {
  test('renders if only required params are passed in', () => {
    render(
      <Skills content={params} />
    )
  
    expect(screen.getByTestId('Mobile')).toMatchSnapshot()
    expect(screen.getByTestId('NotMobile')).toMatchSnapshot()
    screen.getAllByTestId('Title').forEach((title) => {
      expect(title).toHaveTextContent(params.title)
    })
    screen.getAllByTestId('Description').forEach((desc) => {
      expect(desc).toHaveTextContent(params.description[0])
    })


    expect(screen.getAllByTestId('Tile')).toHaveLength(9)
    expect(screen.getByTestId('Row 1').childNodes).toHaveLength(3)
    expect(screen.getByTestId('Row 2').childNodes).toHaveLength(5)
    expect(screen.getByTestId('Row 3').childNodes).toHaveLength(3)
    expect(Button).toHaveBeenCalled()

    for(let i = 0; i < 9; i++) {
      const item = params.subSection.skills[i]
      expect(screen.getAllByRole('link', { name: item.label })).toHaveLength(2)
      screen.getAllByRole('link', { name: item.label }).forEach(link => {
        expect(link).toHaveAttribute('href', item.url)
      })
      screen.getAllByRole('img', { name: item.label }).forEach(image => {
        expect(image).toHaveAttribute('alt', item.label)
      })
    }
  })
  
  test('does not render if less than 9 skills are entered', () => {
    const lessThan9Skills = {
      key: "about",
      label: "Skills",
      inMenu: true,
      title: "test title",
      description: [
        "test contents 1",
        "test contents 2"
      ],
      subSection: {
        skills: [
          {
            key: "github",
            label: "Test Skill 1",
            url: "https://test.skill.one/"
          },
          {
            key: "github",
            label: "Test Skill 2",
            url: "https://test.skill.two/"
          }
        ]
      }
    }

    render(<Skills content={lessThan9Skills} />)
    expect(screen.queryAllByTestId('Mobile')).toHaveLength(0)
    expect(screen.queryAllByTestId('NotMobile')).toHaveLength(0)
  })
})
