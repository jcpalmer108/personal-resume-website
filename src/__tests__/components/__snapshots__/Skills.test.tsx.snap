// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills renders if only required params are passed in 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  padding-bottom: 40px;
}

.c1 {
  padding: 15px 15px 30px 15px;
  max-width: 250px;
}

.c3 {
  height: 50px;
  width: 50px;
  object-fit: cover;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 50px;
  height: 50px;
  border: 1px solid #c6c6c6;
  padding: 30px;
  margin-right: 13px;
}

.c4 {
  font-family: Gilroy-Regular;
  margin: 0px;
  padding: 25px 15px 40px 15px;
}

.c5 {
  padding: 0px;
}

@media (min-width:720px) {
  .c0 {
    display: none;
  }
}

@media (min-width:720px) {
  .c1 {
    max-width: none;
    padding-top: calc(7% - 25px);
    padding-left: 25px;
    position: absolute;
    width: calc(60% - 50px);
    max-width: 500px;
    overflow: hidden;
  }
}

@media (min-width:1024px) {
  .c1 {
    padding-top: 45px;
    padding-left: 60px;
    width: 600px;
  }
}

@media (min-width:720px) {
  .c3 {
    height: 70px;
    width: 70px;
  }
}

@media (min-width:1024px) {
  .c3 {
    height: 100px;
    width: 100px;
  }
}

@media (min-width:720px) {
  .c2 {
    padding: 0px;
    position: absolute;
    border: none;
    margin-right: 0px;
    padding-bottom: 0px;
    max-width: 230px;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    width: 20%;
    padding-top: 7%;
    max-height: 200px;
  }
}

@media (min-width:1024px) {
  .c2 {
    padding-top: 8%;
  }
}

@media (min-width:1155px) {
  .c2 {
    padding-top: 90px;
  }
}

@media (min-width:720px) {
  .c4 {
    padding: 20px 0px 40px 0px;
  }
}

@media (min-width:1155px) {
  .c4 {
    padding: 55px 60px 30px 60px;
  }
}

@media (min-width:720px) {

}

@media (min-width:720px) {

}

@media (min-width:720px) {

}

@media (min-width:1155px) {

}

@media (min-width:720px) {

}

@media (min-width:1155px) {

}

@media (min-width:720px) {

}

@media (min-width:1155px) {

}

@media (min-width:720px) {

}

@media (min-width:1155px) {

}

@media (min-width:720px) {

}

@media (min-width:1155px) {

}

@media (min-width:1155px) {
  .c5 {
    padding: 0px 60px;
  }
}

@media (min-width:1155px) {

}

@media (min-width:1155px) {

}

@media (min-width:720px) {

}

@media (min-width:1024px) {

}

@media (min-width:720px) {

}

@media (min-width:1024px) {

}

@media (min-width:720px) {

}

@media (min-width:1024px) {

}

@media (min-width:720px) {

}

@media (min-width:1024px) {

}

<div
  class="c0"
  data-testid="Mobile"
>
  <h2
    class="c1"
    data-testid="Title"
  >
    test title
  </h2>
  <div
    class=" marquee-container"
    style="--pause-on-hover: running; --pause-on-click: running;"
  >
    <div
      class="overlay"
      style="--gradient-color: rgba(255, 255, 255, 1), rgba(255, 255, 255, 0); --gradient-width: 200px;"
    />
    <div
      class="marquee"
      style="--play: running; --direction: normal; --duration: 0s; --delay: 0s; --iteration-count: infinite;"
    >
      <a
        class="c2"
        href="https://test.skill.one/"
        id="github"
      >
        <img
          alt="Test Skill 1"
          class="c3"
          src="[object Object]"
        />
      </a>
      <a
        class="c2"
        href="https://test.skill.two/"
        id="github"
      >
        <img
          alt="Test Skill 2"
          class="c3"
          src="[object Object]"
        />
      </a>
      <a
        class="c2"
        href="https://test.skill.three/"
        id="github"
      >
        <img
          alt="Test Skill 3"
          class="c3"
          src="[object Object]"
        />
      </a>
      <a
        class="c2"
        href="https://test.skill.four/"
        id="github"
      >
        <img
          alt="Test Skill 4"
          class="c3"
          src="[object Object]"
        />
      </a>
      <a
        class="c2"
        href="https://test.skill.five/"
        id="github"
      >
        <img
          alt="Test Skill 5"
          class="c3"
          src="[object Object]"
        />
      </a>
      <a
        class="c2"
        href="https://test.skill.six/"
        id="github"
      >
        <img
          alt="Test Skill 6"
          class="c3"
          src="[object Object]"
        />
      </a>
      <a
        class="c2"
        href="https://test.skill.seven/"
        id="github"
      >
        <img
          alt="Test Skill 7"
          class="c3"
          src="[object Object]"
        />
      </a>
      <a
        class="c2"
        href="https://test.skill.eight/"
        id="github"
      >
        <img
          alt="Test Skill 8"
          class="c3"
          src="[object Object]"
        />
      </a>
      <a
        class="c2"
        href="https://test.skill.nine/"
        id="github"
      >
        <img
          alt="Test Skill 9"
          class="c3"
          src="[object Object]"
        />
      </a>
    </div>
    <div
      class="marquee"
      style="--play: running; --direction: normal; --duration: 0s; --delay: 0s; --iteration-count: infinite;"
    >
      <a
        class="c2"
        href="https://test.skill.one/"
        id="github"
      >
        <img
          alt="Test Skill 1"
          class="c3"
          src="[object Object]"
        />
      </a>
      <a
        class="c2"
        href="https://test.skill.two/"
        id="github"
      >
        <img
          alt="Test Skill 2"
          class="c3"
          src="[object Object]"
        />
      </a>
      <a
        class="c2"
        href="https://test.skill.three/"
        id="github"
      >
        <img
          alt="Test Skill 3"
          class="c3"
          src="[object Object]"
        />
      </a>
      <a
        class="c2"
        href="https://test.skill.four/"
        id="github"
      >
        <img
          alt="Test Skill 4"
          class="c3"
          src="[object Object]"
        />
      </a>
      <a
        class="c2"
        href="https://test.skill.five/"
        id="github"
      >
        <img
          alt="Test Skill 5"
          class="c3"
          src="[object Object]"
        />
      </a>
      <a
        class="c2"
        href="https://test.skill.six/"
        id="github"
      >
        <img
          alt="Test Skill 6"
          class="c3"
          src="[object Object]"
        />
      </a>
      <a
        class="c2"
        href="https://test.skill.seven/"
        id="github"
      >
        <img
          alt="Test Skill 7"
          class="c3"
          src="[object Object]"
        />
      </a>
      <a
        class="c2"
        href="https://test.skill.eight/"
        id="github"
      >
        <img
          alt="Test Skill 8"
          class="c3"
          src="[object Object]"
        />
      </a>
      <a
        class="c2"
        href="https://test.skill.nine/"
        id="github"
      >
        <img
          alt="Test Skill 9"
          class="c3"
          src="[object Object]"
        />
      </a>
    </div>
  </div>
  <p
    class="c4"
    data-testid="Description"
  >
    test contents 1
  </p>
  <div
    class="c5"
  />
</div>
`;

exports[`Skills renders if only required params are passed in 2`] = `
.c3 {
  padding: 15px 15px 30px 15px;
  max-width: 250px;
}

.c6 {
  height: 50px;
  width: 50px;
  object-fit: cover;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 50px;
  height: 50px;
  border: 1px solid #c6c6c6;
  padding: 30px;
  margin-right: 13px;
}

.c9 {
  font-family: Gilroy-Regular;
  margin: 0px;
  padding: 25px 15px 40px 15px;
}

.c0 {
  display: none;
}

.c8 {
  height: 0px;
  width: 60%;
  max-width: 600px;
  display: block;
}

.c11 {
  height: 0px;
  width: 60%;
  max-width: 600px;
  display: block;
}

.c10 {
  padding: 0px;
}

@media (min-width:720px) {

}

@media (min-width:720px) {
  .c3 {
    max-width: none;
    padding-top: calc(7% - 25px);
    padding-left: 25px;
    position: absolute;
    width: calc(60% - 50px);
    max-width: 500px;
    overflow: hidden;
  }
}

@media (min-width:1024px) {
  .c3 {
    padding-top: 45px;
    padding-left: 60px;
    width: 600px;
  }
}

@media (min-width:720px) {
  .c6 {
    height: 70px;
    width: 70px;
  }
}

@media (min-width:1024px) {
  .c6 {
    height: 100px;
    width: 100px;
  }
}

@media (min-width:720px) {
  .c5 {
    padding: 0px;
    position: absolute;
    border: none;
    margin-right: 0px;
    padding-bottom: 0px;
    max-width: 230px;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    width: 20%;
    padding-top: 7%;
    max-height: 200px;
  }
}

@media (min-width:1024px) {
  .c5 {
    padding-top: 8%;
  }
}

@media (min-width:1155px) {
  .c5 {
    padding-top: 90px;
  }
}

@media (min-width:720px) {
  .c9 {
    padding: 20px 0px 40px 0px;
  }
}

@media (min-width:1155px) {
  .c9 {
    padding: 55px 60px 30px 60px;
  }
}

@media (min-width:720px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (min-width:720px) {
  .c1 {
    width: 100%;
    border-collapse: collapse;
  }
}

@media (min-width:720px) {
  .c8 {
    display: none;
    padding: 25px;
    height: 100%;
    max-width: none;
    width: calc(100% - 50px);
  }
}

@media (min-width:1155px) {
  .c8 {
    display: block;
    position: relative;
    height: 0px;
    padding: 0px;
    width: 100%;
  }
}

@media (min-width:720px) {
  .c11 {
    display: block;
    padding: 25px;
    height: 100%;
    max-width: none;
    width: calc(100% - 50px);
  }
}

@media (min-width:1155px) {
  .c11 {
    display: none;
    position: relative;
    height: 0px;
    padding: 0px;
    width: 100%;
  }
}

@media (min-width:720px) {
  .c2 {
    border-width: 1px;
    border-left: none;
    border-top: none;
    border-bottom: solid;
    border-right: solid;
    border-color: #c6c6c6;
    width: 20%;
    height: 0px;
    padding-bottom: 20%;
  }
}

@media (min-width:1155px) {
  .c2:first-child {
    border-left: none;
  }

  .c2:last-child {
    border-right: none;
  }
}

@media (min-width:720px) {
  .c4 {
    border-width: 1px;
    border-left: solid;
    border-top: solid;
    border-bottom: solid;
    border-right: solid;
    border-color: #c6c6c6;
    width: 20%;
    height: 0px;
    padding-bottom: 20%;
  }
}

@media (min-width:1155px) {
  .c4:first-child {
    border-left: none;
  }

  .c4:last-child {
    border-right: none;
  }
}

@media (min-width:720px) {
  .c7 {
    border-width: 1px;
    border-left: solid;
    border-top: solid;
    border-bottom: none;
    border-right: none;
    border-color: #c6c6c6;
    width: 20%;
    height: 0px;
    padding-bottom: 20%;
  }
}

@media (min-width:1155px) {
  .c7:first-child {
    border-left: none;
  }

  .c7:last-child {
    border-right: none;
  }
}

@media (min-width:1155px) {
  .c10 {
    padding: 0px 60px;
  }
}

@media (min-width:1155px) {

}

@media (min-width:1155px) {

}

@media (min-width:720px) {

}

@media (min-width:1024px) {

}

@media (min-width:720px) {

}

@media (min-width:1024px) {

}

@media (min-width:720px) {

}

@media (min-width:1024px) {

}

@media (min-width:720px) {

}

@media (min-width:1024px) {

}

<div
  class="c0"
  data-testid="NotMobile"
>
  <table
    class="c1"
  >
    <tbody>
      <tr
        data-testid="Row 1"
      >
        <td
          class="c2"
          colspan="3"
        >
          <h2
            class="c3"
            data-testid="Title"
          >
            test title
          </h2>
        </td>
        <td
          class="c4"
        >
          <a
            class="c5"
            data-testid="Tile"
            href="https://test.skill.one/"
            id="github"
          >
            <img
              alt="Test Skill 1"
              class="c6"
              src="[object Object]"
            />
          </a>
        </td>
        <td
          class="c4"
        >
          <a
            class="c5"
            data-testid="Tile"
            href="https://test.skill.two/"
            id="github"
          >
            <img
              alt="Test Skill 2"
              class="c6"
              src="[object Object]"
            />
          </a>
        </td>
      </tr>
      <tr
        data-testid="Row 2"
      >
        <td
          class="c4"
        >
          <a
            class="c5"
            data-testid="Tile"
            href="https://test.skill.three/"
            id="github"
          >
            <img
              alt="Test Skill 3"
              class="c6"
              src="[object Object]"
            />
          </a>
        </td>
        <td
          class="c4"
        >
          <a
            class="c5"
            data-testid="Tile"
            href="https://test.skill.four/"
            id="github"
          >
            <img
              alt="Test Skill 4"
              class="c6"
              src="[object Object]"
            />
          </a>
        </td>
        <td
          class="c4"
        >
          <a
            class="c5"
            data-testid="Tile"
            href="https://test.skill.five/"
            id="github"
          >
            <img
              alt="Test Skill 5"
              class="c6"
              src="[object Object]"
            />
          </a>
        </td>
        <td
          class="c4"
        >
          <a
            class="c5"
            data-testid="Tile"
            href="https://test.skill.six/"
            id="github"
          >
            <img
              alt="Test Skill 6"
              class="c6"
              src="[object Object]"
            />
          </a>
        </td>
        <td
          class="c4"
        >
          <a
            class="c5"
            data-testid="Tile"
            href="https://test.skill.seven/"
            id="github"
          >
            <img
              alt="Test Skill 7"
              class="c6"
              src="[object Object]"
            />
          </a>
        </td>
      </tr>
      <tr
        data-testid="Row 3"
      >
        <td
          class="c4"
        >
          <a
            class="c5"
            data-testid="Tile"
            href="https://test.skill.eight/"
            id="github"
          >
            <img
              alt="Test Skill 8"
              class="c6"
              src="[object Object]"
            />
          </a>
        </td>
        <td
          class="c4"
        >
          <a
            class="c5"
            data-testid="Tile"
            href="https://test.skill.nine/"
            id="github"
          >
            <img
              alt="Test Skill 9"
              class="c6"
              src="[object Object]"
            />
          </a>
        </td>
        <td
          class="c7"
          colspan="3"
        >
          <div
            class="c8"
          >
            <p
              class="c9"
              data-testid="Description"
            >
              test contents 1
            </p>
            <div
              class="c10"
            />
          </div>
        </td>
      </tr>
    </tbody>
  </table>
  <div
    class="c11"
  >
    <p
      class="c9"
      data-testid="Description"
    >
      test contents 1
    </p>
    <div
      class="c10"
    />
  </div>
</div>
`;
